### ISSUE TITLE
Text Input word boundaries are inconvenient (unexpected Ctrl+Backspace behavior)

### ISSUE THREAD

**Issue by Mattwmaster58** [reduf/search#30]
When working with path strings, I frequently use `Ctrl+Backspace` to remove a node from the path. However, in all the input widgets, the word boundaries are inconsistent with the convention of `\` being a word boundary. So instead of removing a node (usually), it removes the entire path.

eg `C:\Data\my\data` (`Ctrl+Backspace`) -> ` `

instead of the expected  `C:\Data\my\data` (`Ctrl+Backspace`) -> `C:\Data\my\`

**Comment by reduf** [comment]
I agree, I'm not sure how much control I have there, since Imgui takes care of that, but I will have a look.

**Comment by reduf** [comment]
Yeah, [this](https://github.com/ocornut/imgui/blob/b902fa4c4490ddffb713f8153d1a65e1940e58d1/imgui_widgets.cpp#L3781-L3797) is the logic, I don't think there is currently a way to customize it.

**Comment by Mattwmaster58** [comment]
Looks like the imgui issue was fixed! Now it's a matter of waiting for a release and updating deps here?

**Comment by Mattwmaster58** [comment]
This has landed in imgui, I think we just need to wait for imgui-rs to be updated? https://github.com/imgui-rs/imgui-rs/issues/834

**Comment by ldufr** [comment]
Thanks for the update. `imgui-rs` hasn't updated yet as far as I can tell. The project seems a bit abandoned...
